pkg/cluster/reconcile.go

```diff
@@ -93,10 +93,10 @@ func (c *Cluster) reconcileMembers(running etcdutil.MemberSet) error {
 		return c.disasterRecovery(L)
 	}
 
-	c.logger.Infof("replacing one dead member with a new member")
-
-	if err := c.removeDeadMember(c.members.Diff(L).PickOne()); err != nil {
-		return err
+	c.logger.Infof("removing one dead member")
+	// remove dead members that doesn't have any running pods before doing resizing.
+	if c.members.Diff(L).Size() > 0 {
```

Comment:
I mean the following:
```
	return c.removeDeadMember(c.members.Diff(L).PickOne())
```

Because if we come to this point, it means that we must have dead members and must have majority alive.